pkg/cluster/node-finder.go

```diff
@@ -1,17 +1,19 @@
 package cluster
 
 import (
+	"context"
 	"fmt"
 	"net"
 	"os"
 	"strings"
+	"time"
 )
 
 // Nodefinders have to return all ips/addr of the serf nodes available on startup
 // the minimum is to return one ip/addr for serf to form a cluster.
 type NodeFinder interface {
-	GetNodes() ([]string, error)
-	GetAddr() (string, error)
+	GetNodes(ctx context.Context) ([]string, error)
+	GetAddr(ctx context.Context, nodeID string) (string, error)
```

Comment:
By adding the nodeID argument I think the purpose of GetAddr has changed pretty significantly. The original purpose was to get the addr of the self. Why was this changed?

Reply:
It looks like both implementations of GetAddr discard this argument as well, so that seems extra strange.